File: app/src/main/java/com/dynamo/spacex/SpaceXItemViewHolder.java
package com.dynamo.spacex;

import androidx.recyclerview.widget.RecyclerView;

import com.dynamo.spacex.databinding.ItemSpacexViewHolderBinding;
import com.dynamo.spacex.viewmodels.SpaceXListViewModel;
import com.dynamo.spacex.viewmodels.SpaceXViewModel;

public class SpaceXItemViewHolder extends RecyclerView.ViewHolder {

    private final ItemSpacexViewHolderBinding itemSpacexViewHolderBinding;

    public SpaceXItemViewHolder(ItemSpacexViewHolderBinding itemSpacexViewHolderBinding) {
        super(itemSpacexViewHolderBinding.getRoot());
        this.itemSpacexViewHolderBinding = itemSpacexViewHolderBinding;
    }

    public void bind(SpaceXViewModel spaceXViewModel, SpaceXListViewModel spaceXListViewModel) {
        itemSpacexViewHolderBinding.setViewModel(spaceXViewModel);
        itemSpacexViewHolderBinding.setListViewModel(spaceXListViewModel);
        itemSpacexViewHolderBinding.executePendingBindings();
    }
}

File: app/src/main/java/com/dynamo/spacex/SpaceXItemViewHolder.kt
package com.dynamo.spacex

import androidx.recyclerview.widget.RecyclerView
import com.dynamo.spacex.databinding.ItemSpacexViewHolderBinding
import com.dynamo.spacex.viewmodels.SpaceXListViewModel
import com.dynamo.spacex.viewmodels.SpaceXViewModel

class SpaceXItemViewHolder (private val itemSpacexViewHolderBinding: ItemSpacexViewHolderBinding): RecyclerView.ViewHolder (itemSpacexViewHolderBinding.root) {

    fun bind(spaceXViewModel: SpaceXViewModel, spaceXListViewModel: SpaceXListViewModel) {
        itemSpacexViewHolderBinding.viewModel = spaceXViewModel
        itemSpacexViewHolderBinding.listViewModel = spaceXListViewModel
        itemSpacexViewHolderBinding.executePendingBindings()
    }
}